pilot/pkg/model/context.go

```diff
@@ -859,14 +859,20 @@ func (node *Proxy) SetServiceInstances(serviceDiscovery ServiceDiscovery) {
 	node.ServiceInstances = instances
 }
 
-// SetWorkloadLabels will set the node.Metadata.Labels only when it is nil.
+// SetWorkloadLabels will set the node.Metadata.Labels with priority:
+// 1. Proxy workload labels.
+// 2. node meta labels.
 func (node *Proxy) SetWorkloadLabels(env *Environment) {
```

Comment:
So is this merging pod labels and metadata labels giving preference to pod labels - which makes sense but the comment is confusing. May be we should say it merges both and gives preference to proxy label.